socorro/siglists/__init__.py

```diff
@@ -23,27 +24,46 @@
                 'nsILocalFile**)'
             ) in x
         ),
-    ),
+    ],
 }
 
 
+class BadRegularExpressionLineError(Exception):
+    """Raised when a file contains an invalid regular expression."""
+
+
 def _get_file_content(source):
     """Return a tuple, each value being a line of the source file.
 
     Remove empty lines and comments (lines starting with a '#').
     """
     filepath = os.path.join(_DIRECTORY, source + '.txt')
 
+    lines = []
     with open(filepath, 'r') as f:
-        lines = tuple(
-            line.replace('\n', '') for line in f
-            if line.strip() and not line.strip().startswith('#')
-        )
+        for (i, line) in enumerate(f):
+            line = line.strip()
+
+            if not line or line.startswith('#'):
+                continue
+
+            try:
+                re.compile(line)
```

Comment:
``signature_utilities.py`` joins the lists with '|' and then ``re.compile`` the result. So using ``re.compile`` on individual items covers it well.